### ISSUE TITLE
Flow makes `react/no-unused-prop-types` fail

### ISSUE THREAD

**Issue by laplacesdemon** [gajus/eslint-plugin-flowtype#363]
Apologies if this is not relevant to this project.

I have a component and an arrow function in it. I have a type annotation as in the example below. Flow checker is passing but eslint gives following error: `'label' PropType is defined but prop is never used  react/no-unused-prop-types`. Removing the type annotation, i.e. `{label:string}` fixes the problem with eslint but flow checker gives an error for missing annotation.

```javascript
class SomeComponent extends React.PureComponent<Props> {
  handleLabel = ({ label }: { label: string }) => `${label}:`;
}
```

Version: `"eslint-plugin-flowtype": "^2.50.0",`

**Comment by nathanredblur** [comment]
This happen to me too. :(

**Comment by gajus** [comment]
As the error says, this belongs in https://github.com/yannickcr/eslint-plugin-react/issues.
